doc/user/client.md

```diff
@@ -0,0 +1,411 @@
+// struct pointer so that obj can be updated with the content returned
+// by the API server.
+func (c Client) Update(ctx context.Context, obj runtime.Object) error
+```
+
+```Go
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+	
+	dep := &v1.Deployment{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, dep)
+
+	...
+
+	ctx := context.TODO()
+	dep.Spec.Selector.MatchLabels["is_running"] = "true"
+	err := r.client.Update(ctx, dep)
+
+	...
+}
+```
+
+#### Delete
+
+```Go
+// Delete deletes the given obj from k8s cluster.
+func (c Client) Delete(ctx context.Context, obj runtime.Object, opts ...DeleteOptionFunc) error
+```
+
+```Go
+// DeleteOptionFunc is a function that mutates a DeleteOptions struct.
+type DeleteOptionFunc func(*DeleteOptions)
+
+type DeleteOptions struct {
+    // GracePeriodSeconds is the duration in seconds before the object should be
+    // deleted. Value must be non-negative integer. The value zero indicates
+    // delete immediately. If this value is nil, the default grace period for the
+    // specified type will be used.
+    GracePeriodSeconds *int64
+
+    // Preconditions must be fulfilled before a deletion is carried out. If not
+    // possible, a 409 Conflict status will be returned.
+    Preconditions *metav1.Preconditions
+
+    // PropagationPolicy determined whether and how garbage collection will be
+    // performed. Either this field or OrphanDependents may be set, but not both.
+    // The default policy is decided by the existing finalizer set in the
+    // metadata.finalizers and the resource-specific default policy.
+    // Acceptable values are: 'Orphan' - orphan the dependents; 'Background' -
+    // allow the garbage collector to delete the dependents in the background;
+    // 'Foreground' - a cascading policy that deletes all dependents in the
+    // foreground.
+    PropagationPolicy *metav1.DeletionPropagation
+
+    // Raw represents raw DeleteOptions, as passed to the API server.
+    Raw *metav1.DeleteOptions
+}
+```
+
+```Go
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+	
+	pod := &v1.Pod{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, pod)
+
+	...
+
+	ctx := context.TODO()
+	if pod.Status.Phase == v1.PodUnknown {
+		df := func(opts *client.DeleteOptions) {
+			s := 5
+			opts.GracePeriodSeconds = &s // Delete the pod after 5 seconds.
+		}
+		err := r.client.Delete(ctx, pod, df)
+
+		...
+	}
+
+	...
+}
+```
+
+#### Get
+
+```Go
+// Get retrieves an API object for a given object key from the k8s cluster
+// and stores it in obj.
+func (c Client) Get(ctx context.Context, key ObjectKey, obj runtime.Object) error
+```
+
+```Go
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+	
+	app := &v1alpha1.App{}
+	ctx := context.TODO()
+	name := request.NamespacedName
+	err := r.client.Get(ctx, name, app)
```

Comment:
Get takes [types.NamespacedName](https://github.com/kubernetes-sigs/controller-runtime/blob/master/vendor/k8s.io/apimachinery/pkg/types/namespacedname.go#L31-L34) as input. Not name.
```Go
err := r.client.Get(context.TODO(), request.NamespacedName, app)
```

And probably move the section for `Get` before `Delete` and `Update` since they're both using Get in their examples.

Reply:
`name` is `request.NamespacedName`, I moved it to its own variable for clarity.

Reply:
Yeah the variable `name` makes it seem like that parameter is just the name, when it's not. 
So maybe just pass it in directly. That should still be clear enough imo.